sdk/eventhub/Microsoft.Azure.EventHubs.Processor/src/AzureStorageCheckpointLeaseManager.cs

```diff
@@ -406,14 +406,8 @@ async Task<bool> AcquireLeaseCoreAsync(AzureBlobLease lease)
                     await this.RenewLeaseCoreAsync(lease).ConfigureAwait(false);
                 }
 
-                // Update owner in the metadata first since clients get ownership information by looking at metadata.
+                // Update owner in the metadata and upload the lease content.
                 lease.Blob.Metadata[MetaDataOwnerName] = lease.Owner;
-                await lease.Blob.SetMetadataAsync(
-                    AccessCondition.GenerateLeaseCondition(lease.Token),
-                    this.defaultRequestOptions,
-                    this.operationContext).ConfigureAwait(false);
-
-                // Then update deserialized lease content.
                 await leaseBlob.UploadTextAsync(
```

Comment:
So the upload also sets the metadata, it would seem...  no?

Reply:
Exactly. Also documented here - https://docs.microsoft.com/en-us/rest/api/storageservices/Put-Blob?redirectedfrom=MSDN